src/docbuilder/rustwide_builder.rs

```diff
@@ -514,16 +539,20 @@ impl RustwideBuilder {
                 .run()
                 .is_ok()
         });
-        // If we're passed a default_target which requires a cross-compile,
-        // cargo will put the output in `target/<target>/doc`.
-        // However, if this is the default build, we don't want it there,
-        // we want it in `target/doc`.
-        if target != HOST_TARGET && is_default_target {
-            // mv target/$target/doc target/doc
+        // If we're passed the host target, cargo will put the output in `target/doc`.
+        // However, we don't want it there, we want it in `target/$HOST_TARGET/doc`.
+        // See comments to `build_package` for an explanation of why.
+        if target == HOST_TARGET {
```

Comment:
So Cargo actually doesn't *quite* behave this way, I think, if I follow this comment. In particular, so long as we pass `--target`, target-specific artifacts will always go into `target/$target/` I think. But host-artifacts (e.g. build scripts and proc macros) go into `target/`.

Can you make sure that this doesn't require changes to the code here? I'm not 100% confident myself; it looks right, though, I think.

Reply:
Ah yes good point - the reason files are going into `target/doc` is because we specifically don't pass `--target` for the HOST_TARGET. I'll add that as a comment.